### PR TITLE
AC-6375: Rebuild person profile page using front-end directory repo

### PR DESCRIPTION
**Pull request by VincentHokie**
#### Changes introduced in [AC-6375](https://masschallenge.atlassian.net/browse/AC-6375)
- add graphql types and logic to allow us get information about an entrepreneur

#### How to test
- checkout this branch
- run impact
- visit `localhost/graphql`
- enter this in the query field (top left)
```
query getEntrepreneurProfile($id: Int!) {
  entrepreneurProfile(id: $id) {
    title
    user {
      id
      firstName
      lastName
      email
    }
    startups {
      id
      name
      shortPitch
      highResolutionLogo
      program {
        family
        year
      }
    }
    profile {
      phone
      linkedInUrl
      facebookUrl
      twitterHandle
      personalWebsiteUrl
      imageUrl
    }
    profile {
      currentProgram
    }
    
  }
}
```

- now enter the following in the bottom left with a startup team member id and see results on the right column
```
{"id": <startup-team-member-id>}
```

### CODE REVIEW

**Comment by VincentHokie** [comment]
I would not merge this PR until the rest of the PRs are approved and ready
- https://github.com/masschallenge/directory/pull/129
- https://github.com/masschallenge/accelerate/pull/2056
- https://github.com/masschallenge/django-accelerator/pull/138

a change in one could lead to a ripple effect in the rest

**Comment by mkiterian** [comment]
The url defined on the ticket was `/people/{{ user_id }}/`, is there a reason we've switched to `/people/{{ team_member_id }}/`?

**Comment by VincentHokie** [comment]
@mkiterian 

Yes, sorry, this is a conversation that happened upon the DMs. 
First, the existing (soon to be legacy) page uses the team member id.
Also, the user title and startup info is tied to the team member object. The user-startup_team_member relationship is a one-to-many one. So that means for every user_id we could have multiple team members.

As far as the title, if every team member profile has a title we end up with a situation where we need to choose which title to display (we could end up choosing a wrong one). Using the team_member_id we don't have to deal with that dynamic.

As a side-note, in my opinion, moving to a user_id in the url may require us moving the title field from the team_member profile to the entrepreneur profile.

File: web/impact/impact/graphql/types/startup_type.py
Comment (mkiterian) [inline web/impact/impact/graphql/types/startup_type.py]: If a startup has been a finalist in more than one program, would this get the right program?

File: web/impact/impact/graphql/types/startup_team_member_type.py
Comment (VincentHokie) [reply web/impact/impact/graphql/types/startup_team_member_type.py]: I actually haven't figured out how to send back an object, any ideas? Could you point me to a good resource/ give me pointers

File: web/impact/impact/graphql/types/startup_team_member_type.py
Comment (VincentHokie) [reply web/impact/impact/graphql/types/startup_team_member_type.py]: this is true, changing a pushing

Comment (VincentHokie) [reply web/impact/impact/graphql/types/startup_team_member_type.py]: done

File: web/impact/impact/graphql/types/startup_type.py
Comment (VincentHokie) [reply web/impact/impact/graphql/types/startup_type.py]: ordered this from newest to oldest based on creation_date, now for sure it will get the most recent startup status

File: web/impact/impact/graphql/types/startup_team_member_type.py
Comment (mkiterian) [reply web/impact/impact/graphql/types/startup_team_member_type.py]: This should solve it: `profile = graphene.Field(EntrepreneurProfileType)`

**Review by mkiterian — DISMISSED** [review]

File: web/impact/impact/graphql/query.py
Comment (VincentHokie) [reply web/impact/impact/graphql/query.py]: thanks for this

Comment (VincentHokie) [reply web/impact/impact/graphql/query.py]: removed

File: web/impact/impact/graphql/types/startup_type.py
Comment (kitaka) [inline web/impact/impact/graphql/types/startup_type.py]: @VincentHokie can be removed.

**Review by kitaka — CHANGES_REQUESTED** [review]

File: web/impact/impact/graphql/types/startup_team_member_type.py
Comment (VincentHokie) [reply web/impact/impact/graphql/types/startup_team_member_type.py]: done, arigato

File: web/impact/impact/graphql/types/startup_team_member_type.py
Comment (VincentHokie) [reply web/impact/impact/graphql/types/startup_team_member_type.py]: done, arigato

File: web/impact/impact/graphql/types/startup_type.py
Comment (VincentHokie) [reply web/impact/impact/graphql/types/startup_type.py]: done, arigato

File: web/impact/impact/graphql/types/program_type.py
Comment (VincentHokie) [reply web/impact/impact/graphql/types/program_type.py]: done, arigato

File: web/impact/impact/graphql/types/startup_type.py
Comment (VincentHokie) [reply web/impact/impact/graphql/types/startup_type.py]: done, arigato

File: web/impact/impact/tests/test_graphql.py
Comment (VincentHokie) [reply web/impact/impact/tests/test_graphql.py:159]: not until now to be honest, but reading through this..with the exception of the snapshot tests (which I'm not a huge fan of) the test cases in the docs are doing roughly what we are currently doing. I don't see major gains here

**Review by kitaka — CHANGES_REQUESTED** [review]

File: Makefile
Comment (VincentHokie) [reply Makefile]: this was undoing a change I had made by mistake, then a PR with a change I was undoing was merged in. I'll correct this.

**Comment by VincentHokie** [comment]
I honestly don't see a win with the constant you're suggesting [here](https://github.com/masschallenge/impact-api/pull/261#discussion_r259314058), it's used once in the whole impact codebase. So, I'm inclined to let that be.

File: Makefile
Comment (VincentHokie) [reply Makefile]: corrected.

**Review by kitaka — APPROVED** [review]

**Comment by VincentHokie** [comment]
Suggested changes have been made i.e. standardising the data returned by the Expert and Entrepreneur profiles. Changes ready for review.

**Comment by kaymation** [comment]
@mkiterian clear from the code that the changes you've requested have been fixed. dismissing your review to unblock this :)

**Review by kaymation — APPROVED** [review]
QA'd to my confort, works as advertised

**Pull request merged**
